src/r3/atomic-swap/corda/evm-bridge-contracts/src/main/kotlin/com/r3/corda/DefaultEventEncoder.kt

```diff
@@ -0,0 +1,117 @@
+package com.r3.corda.evmbridge
+
+import com.r3.corda.cno.evmbridge.dto.TransactionReceipt
+import com.r3.corda.cno.evmbridge.dto.TransactionReceiptLog
+import net.corda.core.serialization.CordaSerializable
+import org.web3j.abi.DefaultFunctionEncoder
+import org.web3j.crypto.Hash
+import org.web3j.abi.TypeEncoder
+import org.web3j.abi.datatypes.Type
+import org.web3j.abi.datatypes.*
+import org.web3j.abi.datatypes.generated.*
+import org.web3j.utils.Numeric
+import java.io.Serializable
+import java.math.BigInteger
+
+data class Indexed<T>(val indexedValue: T)
+
+class DefaultEventEncoder {
```

Comment:
could we get away with just making this an object? the class has no state and doesn't implement an interface so rather than allocate it somewhere, just make it a singlton